holmes/version.py

```diff
@@ -0,0 +1,178 @@
+    # we are running from an unreleased dev version
+    try:
+        # Get the latest git tag
+        tag = (
+            subprocess.check_output(
+                ["git", "describe", "--tags"], stderr=subprocess.STDOUT, cwd=this_path
+            )
+            .decode()
+            .strip()
+        )
+
+        # Get the current branch name
+        branch = (
+            subprocess.check_output(
+                ["git", "rev-parse", "--abbrev-ref", "HEAD"],
+                stderr=subprocess.STDOUT,
+                cwd=this_path,
+            )
+            .decode()
+            .strip()
+        )
+
+        # Check if there are uncommitted changes
+        status = (
+            subprocess.check_output(
+                ["git", "status", "--porcelain"],
+                stderr=subprocess.STDOUT,
+                cwd=this_path,
+            )
+            .decode()
+            .strip()
+        )
+        dirty = "-dirty" if status else ""
+
+        return f"{tag}-{branch}{dirty}"
+
+    except Exception:
+        pass
+
+    # we are running without git history, but we still might have git archival data (e.g. if we were pip installed)
+    archival_file_path = os.path.join(this_path, ".git_archival.json")
+    if os.path.exists(archival_file_path):
+        try:
+            with open(archival_file_path, "r") as f:
+                archival_data = json.load(f)
+                return f"dev-{archival_data['refs']}-{archival_data['hash-short']}"
+        except Exception:
+            pass
+
+        return "dev-version"
+
+    return "unknown-version"
+
+
+@cache
+def fetch_holmes_info() -> Optional[HolmesInfo]:
+    """Fetch latest version information from Robusta API"""
+    try:
+        response = requests.get(HOLMES_GET_INFO_URL, timeout=TIMEOUT)
+        response.raise_for_status()
+        result = response.json()
+        return HolmesInfo(**result)
+    except Exception:
+        return None
+
+
+def check_version() -> VersionCheckResult:
+    """
+    Centralized version checking logic.
+    Returns complete version check result with message.
+    """
+    current_version = get_version()
+    holmes_info = fetch_holmes_info()
+
+    # Default to latest if we can't determine
+    if not holmes_info or not holmes_info.latest_version or not current_version:
+        return VersionCheckResult(
+            is_latest=True, current_version=current_version or "unknown"
+        )
+
+    # Dev versions are considered latest
+    if current_version.startswith("dev-"):
+        return VersionCheckResult(
+            is_latest=True,
+            current_version=current_version,
+            latest_version=holmes_info.latest_version,
+        )
+
+    # Check if current version starts with latest version
+    is_latest = current_version.startswith(holmes_info.latest_version)
+
+    update_message = None
+    if not is_latest:
+        update_message = f"Update available: v{holmes_info.latest_version} (current: {current_version})"
```

Comment:
nit: should we remove the prefix v to let the latest and current version pattern match?